libdisni/src/verbs/com_ibm_disni_rdma_verbs_impl_NativeDispatcher.cpp

```diff
@@ -700,6 +704,68 @@ JNIEXPORT jint JNICALL Java_com_ibm_disni_rdma_verbs_impl_NativeDispatcher__1mod
 	return ret;
 }
 
+/*
+ * Class:     com_ibm_jverbs_nat_NativeDispatcher
+ * Method:    _queryOdpSupport
+ * Signature: (J)I
+ */
+JNIEXPORT jint JNICALL Java_com_ibm_disni_rdma_verbs_impl_NativeDispatcher__1queryOdpSupport
+  (JNIEnv *env, jobject obj, jlong id){
+       jint ret = -1;
+       #ifdef HAVE_ODP_MR_PREFETCH
+           struct rdma_cm_id *cm_listen_id = (struct rdma_cm_id *)id;
+           struct ibv_device_attr_ex dev_attr;
+           ret = ibv_query_device_ex(cm_listen_id->verbs, NULL, &dev_attr);
+           if (ret == 0) {
+            if (dev_attr.odp_caps.general_caps & IBV_ODP_SUPPORT &&
+                dev_attr.odp_caps.per_transport_caps.rc_odp_caps & IBV_ODP_SUPPORT_SEND &&
+                dev_attr.odp_caps.per_transport_caps.rc_odp_caps & IBV_ODP_SUPPORT_RECV &&
+                dev_attr.odp_caps.per_transport_caps.rc_odp_caps & IBV_ODP_SUPPORT_READ &&
+                dev_attr.odp_caps.per_transport_caps.rc_odp_caps & IBV_ODP_SUPPORT_ATOMIC) {
+                           log("j2c::queryOdpSupport: supported");
```

Comment:
Why not pass this to the Java code and let the user know which operations support ODP?

Reply:
Would be a bit different semantic from other methods (jint returns exit code mostly), but here would need to handle 2 numbers in java code then (exit code OR rc_odp_caps). But yes, could change this if you need: if return is -1, then no ODP supported otherwise do bitmask.

Reply:
I would prefer it since it also looks simple enough to implement but as I said I'm also fine with merging as is if nobody else thinks that we want this.

Reply:
Done, now `queryOdpSupport` returns odp_caps.per_transport_caps.rc_odp_caps property if ODP is supported, -1 otherwise.